cli/flox-rust-sdk/src/models/environment/mod.rs

```diff
@@ -328,10 +319,27 @@ pub struct DotFlox {
 }
 
 impl DotFlox {
-    pub fn open(path: impl AsRef<Path>) -> Result<Self, EnvironmentError> {
+    /// Open `<parent_path>/.flox` as a [DotFlox] directory
+    ///
+    /// This method attempts to find a `.flox` directory in the specified parent path,
+    /// and open it as a [DotFlox] directory.
+    /// If the directory is not found, an [EnvironmentError::DotFloxNotFound] is returned.
+    pub fn open_default_in(parent_path: impl AsRef<Path>) -> Result<Self, EnvironmentError> {
+        let dot_flox_path = parent_path.as_ref().join(DOT_FLOX);
+        let dot_flox_path = CanonicalPath::new(&dot_flox_path)
+            .map_err(|_| EnvironmentError::DotFloxNotFound(dot_flox_path))?;
+
+        Self::open(&dot_flox_path)
+    }
+
+    /// Open the specified path as a [DotFlox] directory
+    ///
+    /// This method is used to open an existing [DotFlox] directory.
+    pub(crate) fn open(dot_flox_path: &CanonicalPath) -> Result<Self, EnvironmentError> {
```

Comment:
nonblocking: we only ever use this in `open_default_in`, and we assume when displaying errors that we always use `open_default_in`, so not sure we should keep this method around

Reply:
I think we do also use it to open "active environments"

Reply:
Ah well, we actually dont, the struct is deserialized into a dotflox directly.
That means keeping this only makes academic sense but is without purpose within flox right now indeed.